brpc-java-core/src/main/java/com/baidu/brpc/client/handler/ClientWorkTask.java

```diff
@@ -39,8 +50,18 @@
 
     @Override
     public void run() {
-        Response response;
 
+        // 只有server push协议下，有可能受到request类型
+        if (protocol instanceof ServerPushProtocol) {
+            // 区分类型
+            SPHead spHead = ((ServerPushPacket) packet).getSpHead();
```

Comment:
please define the ServerPushPacket interface and DefaultServerPushPacket, since custom push body may not be the SPBody.